src/wagtailtrans/signals.py

```diff
@@ -43,7 +43,7 @@ def synchronize_deletions(sender, instance, **kwargs):
 
     """
     language = getattr(instance, 'language', False)
-    if language and not instance.canonical_page:
```

Comment:
`if language and not instance.canonical_page:` is not the same as  `if language and not instance.is_canonical:` change into `if language and instance.is_canonical:`